proposals/infra/INF-0005-Long-Vector-ExecutionTest-Plan.md

```diff
@@ -0,0 +1,363 @@
+respectively. This boundary will tested for with 32-bit and 64-bit sized values
+as well.
+
+* Most GPUs operate on at least 32-bits at once, so what happens if you use
+  16-bit values and an odd number of elements. Could accesssing the last element
+  expose issues where we could overwrite the next variable is it is assuming
+  alignment?
+
+## High level test design
+
+1. The test will leverage the existing XML infrastructure currently used by the
+   existing execution tests. There are two XML files. This general design
+   pattern exists today in the execution tests.
+
+   * 1st XML: Used to define shader source code and metadata about that shader
+   code. This XML file is parsed using a private class. This private class helps
+   faciliate creation of D3D resources and execution of the shader.
+   * 2nd XML: Describes metadata about the specific test cases. Used by the
+   TAEF infrastructe for [TAEF Data Driven
+   Testing](https://learn.microsoft.com/en-us/windows-hardware/drivers/taef/data-driven-testing)
+2. Test inputs will be hard coded in a c++ header file. This was chosen over
+   definining inputs in the second XML as this is cleaner and easier to parse
+   for different data types. This c++ header method also avoids needing to
+   repeat the data set in the XML for each individual test cast. Inputs will use
+   'value sets' which will typically be much smaller than the desired vector
+   test size. Values will be repeated until the vector is full.
+3. Expected outputs are computed for each test case at run time.
+4. All new long vector test code is factored out into its own files.
+
+## Implementation phases
+
+Do the test work in two simple phases.
+
+1. Implement and validate (locally against WARP) for all test categories.
+2. HLK related work:
+
+* Add a SM 6.9 HLK requirement. Includes updating the HLK requirements doc.
+* Update mm_annotate_shader_op_arith_table.py to annotate the new test cases
+  with HLK GUIDS and requirements
+* Add new tests to HLK playlist
+
+## Shipping
+
+Note that because DXC and the Agility SDK are both undocked from Windows it is
+our normal operating behavior for the HLK tests to become available with a later
+TBD OS release. The good news is that this doesn't prevent the tests from being
+available much earlier in the DXC repo. It just means that they are simply TAEF
+tests in the DXC repo. An HLK test includes an extra level of infrastructure for
+test gating, selection, and result submission for WHQL signing of drivers.
+
+1. Tests will be shared privately with IHVs along with the latest DXC and
+latest Agility SDK for tesing and validation. IHVs will also be able to build
+and run the tests from the public DXC repo themselves. If needed Microsoft can
+share further instructions when the tests are available.
+
+2. The tests will ship with the HLK at a TBD date in a later OS release.
+
+## New HLK Tests
+
+All of the scripts and yaml files mentioned are part of the Microsoft WinTools
```

Comment:
Unsure if we should / need to mention the internal WinTools repo, or if we can leave it out of this spec?